module/gui/LinkDock.py

```diff
@@ -19,38 +19,210 @@
+        self.append = QCheckBox(_("Append URLs"))
+        whatsThis = (self.append.text(), _("Append pasted/dropped text instead of inserting it at the cursor postion."))
+        self.append.setWhatsThis(whatsThisFormat(*whatsThis))
+        whatsThis = (boxLabel.text(), _("One URL (Link) per line.<br>For manual text editing you may want to deactivate") + " '" + self.append.text() + "'.")
+        self.box.setWhatsThis(whatsThisFormat(*whatsThis))
+        self.destQueue     = QRadioButton(_("Queue"))
+        self.destCollector = QRadioButton(_("Collector"))
+        destBtnLayout = QHBoxLayout()
+        destBtnLayout.addWidget(self.destQueue)
+        destBtnLayout.addWidget(self.destCollector)
+        destBtnLayout.addStretch(1)
+        self.destAutoSelect = QCheckBox(_("Select with tab"))
+        
+        self.clear = QPushButton(_("Clear"))
+        whatsThis = (self.clear.text(), _("Clears the URL box."))
+        self.clear.setWhatsThis(whatsThisFormat(*whatsThis))
+        self.filter = QPushButton()
+        
+        self.save = QPushButton(_("Add"))
+        self.save.setIcon(self.dock.appIconSet["add_small"])
+        self.msg = QLabel("ERROR")
+        s = "QWidget { color: crimson; background-color: %s }" % QColor(Qt.gray).name()     # red text color on gray background
+        self.msg.setStyleSheet(s)
+        lsp = self.msg.sizePolicy()
+        lsp.setHorizontalPolicy(QSizePolicy.Ignored)
+        self.msg.setSizePolicy(lsp)
+        hbox = QHBoxLayout()
+        hbox.addWidget(self.msg)
+        hbox.addWidget(self.save)
         
-        layout.addWidget(explanationLabel)
         layout.addWidget(boxLabel)
         layout.addWidget(self.box)
-        layout.addWidget(save)
+        layout.addWidget(self.append)
+        layout.addWidget(self.filter)
+        layout.addWidget(self.clear)
+        layout.addLayout(destBtnLayout)
+        layout.addWidget(self.destAutoSelect)
+        layout.addLayout(hbox)
+        
+        self.adjustSize()
+        self.msg.setFixedHeight(self.save.height())
+        self.slotMsgHide()
         
-        self.connect(save, SIGNAL("clicked()"), self.dock.slotDone)
+        self.connect(self.save, SIGNAL("clicked()"), self.dock.slotDone)
+        self.connect(self.filter, SIGNAL("clicked()"), self.dock.slotFilterBtnClicked)
+        self.connect(self.box, SIGNAL("textChanged()"), self.dock.slotBoxTextChanged)
+        self.connect(self.append, SIGNAL("toggled(bool)"), self.box.slotAppendToggled)
+        self.box.slotAppendToggled(self.append.isChecked())
+        self.connect(self.clear, SIGNAL("clicked()"), self.dock.slotClearBtnClicked)
+    
+    def slotMsgShow(self, msg):
+        self.msg.setText(msg)
+        self.msg.setFixedHeight(self.save.height())
+        self.msg.show()
+        self.save.hide()
+        QTimer.singleShot(2000, self.slotMsgHide)
+    
+    def slotMsgHide(self):
+        self.msg.hide()
+        self.save.show()
+
+class PlainTextEdit(QPlainTextEdit):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['QPlainTextEdit' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)